include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -112,7 +112,7 @@ __brick_walk1(_ForwardIterator __first, _ForwardIterator __last, _Function __f,
 template <class _RandomAccessIterator, class _Function>
 void
 __brick_walk1(_RandomAccessIterator __first, _RandomAccessIterator __last, _Function __f,
-              /*vector=*/::std::true_type) noexcept
+              /*vector=*/::std::true_type)
 {
```

Comment:
So is it the provided function that may throw, or just simply the bracket dereference operator? (within `__simd_walk_1`)

Reply:
It's the body of the functor '__f'' that may throw.  It's several layers up in the stack from this point.